solv/tdnfpackage.c

```diff
@@ -820,6 +820,60 @@ SolvGetPackageId(
     goto cleanup;
 }
 
+uint32_t
+SolvGetPkgChecksumFromId(
+    PSolvSack pSack,
+    uint32_t dwPkgId,
+    int checksumtype,
+    int checksumlen,
+    unsigned char** ppszChecksum)
+{
+
+    uint32_t dwError = 0;
+    const unsigned char* pszTemp = NULL;
+    unsigned char* pszChecksum = NULL;
+    Solvable *pSolv = NULL;
+
+    if(!pSack || !pSack->pPool || !ppszChecksum)
+    {
+        dwError = ERROR_TDNF_INVALID_PARAMETER;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    pSolv = pool_id2solvable(pSack->pPool, dwPkgId);
+    if(!pSolv)
+    {
+        dwError = ERROR_TDNF_NO_DATA;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    pszTemp = solvable_lookup_bin_checksum(pSolv, SOLVABLE_CHECKSUM, &checksumtype);
+
+    if(!pszTemp)
+    {
+        dwError = ERROR_TDNF_NO_DATA;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    dwError = TDNFAllocateMemory(checksumlen, sizeof(unsigned char), (void **)&pszChecksum);
+    BAIL_ON_TDNF_ERROR(dwError);
+
+    memcpy(pszChecksum, pszTemp, checksumlen);
+    *ppszChecksum = pszChecksum;
```

Comment:
Outside this function there is a memory leak not freeing the ppszChecksum memory .
once you return from this function while cleanup TDNFFreePackageInfo is getting called need to add free for ppszChecksum.